bin/reference_tools/update_bed.py

```diff
@@ -0,0 +1,450 @@
+    )
+
+    ensembl_bed_path = Path(f"{out_dir}/exons_hg38_{release}.bed")
+
+    print("Write initial base (ENSEMBL exons)")
+    write_ensembl_bed(out_dir, ensembl_bed_path, release, skip_download, EXON_PAD)
+
+    if len(incl_bed) > 0:
+        for bed_fp in incl_bed:
+            print(f"Additional included BED file: {bed_fp}")
+            suffix = bed_fp
+            append_to_bed(final_bed_path, Path(bed_fp), suffix)
+    else:
+        print("No extra BED files to include")
+
+    append_to_bed(final_bed_path, ensembl_bed_path, f"EXONS-{release}")
+
+    (clinvar_new, new_benign) = read_clinvar(clinvar_vcf_path)
+    (clinvar_old, _old_benign) = read_clinvar(clinvar_vcf_old_path)
+
+    (new_to_add_clinvar_bed_rows, old_to_remove_clinvar_bed_rows) = compare_clinvar(
+        clinvar_new, clinvar_old, final_bed_path, VARIANT_PAD, out_dir, keep_tmp
+    )
+
+    new_to_remove_keys = [bed_row[4] for bed_row in new_to_add_clinvar_bed_rows]
+    old_to_remove_keys = [bed_row[4] for bed_row in old_to_remove_clinvar_bed_rows]
+
+    clinvar_log_path = ensure_new_empty(f"{out_dir}/clinvar_{clinvardate}.log")
+    log_changes(
+        clinvar_log_path,
+        clinvar_old,
+        clinvar_new,
+        set(new_to_remove_keys),
+        set(old_to_remove_keys),
+        new_benign,
+    )
+
+    new_clinvar_to_add_path = Path(f"{out_dir}/new_clinvar_to_add.bed")
+    new_clinvar_to_add_path.write_text(
+        "\n".join(["\t".join(row[0:3]) for row in new_to_add_clinvar_bed_rows]) + "\n"
+    )
+
+    append_to_bed(final_bed_path, new_clinvar_to_add_path, ".")
+    sort_merge_output(out_dir, final_bed_path, keep_tmp)
+
+
+def ensure_new_empty(filepath: str) -> Path:
+    path = Path(filepath)
+    if path.exists():
+        path.unlink()
+    path.write_text("")
+    return path
+
+
+class ClinVarVariant:
+
+    @staticmethod
+    def from_line(vcf_line: str) -> "ClinVarVariant":
+
+        vcf_line = vcf_line.rstrip()
+        fields = vcf_line.split("\t")
+        chrom = fields[0]
+        pos = int(fields[1])
+        ref = fields[3]
+        alt = fields[4]
+        info_str = fields[7]
+
+        info: dict[str, str] = {}
+        for info_field in info_str.split(";"):
+            (key, val) = info_field.split("=")
+            info[key] = val
+
+        return ClinVarVariant(chrom, pos, ref, alt, info)
+
+    def __init__(
+        self, chrom: str, pos: int, ref: str, alt: str, info: dict[str, str] = {}
+    ):
+
+        self.chrom = chrom
+        self.pos = pos
+        self.ref = ref
+        self.alt = alt
+        self.info = info
+
+        self.reason = "Undefined"
+
+        self.CLNDN = (
+            self.info[CLNDN] if self.info.get(CLNDN) is not None else "Undefined"
+        )
+        self.CLNSIG = self.info[CLNSIG] if self.info.get(CLNSIG) else "Undefined"
+
+    def set_reason(self, reason: str):
+        self.reason = reason
```

Comment:
the above is not a requested change, just a tip to save you the future pain of writing getter/setter functions before you really need them

Reply:
Mm, I am familiar with getters and settings. Some like them, some don't. We can discuss at some point how valuable they are when and where.

In this case the class is mostly a container of data with some utils. I think it would add clutter here and not so much benefit here.

When developing, I like to do "lazy evaluation" as well, i.e. not adding functionality until it actually is needed, to avoid adding potentially unneeded complexity.

Thanks for the suggestion!

Reply:
Also on the reread I got what you actually meant here - for the cases where I have written `set` and `get`. Yes, I'll update.